apps/nestjs-api/src/reminder-emails/reminder-emails.service.ts

```diff
@@ -0,0 +1,288 @@
+    if (!userType) {
+      throw new Error('User type is required')
+    }
+
+    const createdDate = new Date()
+    createdDate.setDate(createdDate.getDate() - 14)
+    createdDate.setHours(0, 0, 0, 0) // set time to 00:00
+
+    const endOfCreatedDate = new Date(createdDate)
+    endOfCreatedDate.setHours(23, 59, 59, 999) // set time to 23:59
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': userType,
+      Profile_Status__c: ConnectProfileStatus.DRAFTING_PROFILE,
+      CreatedDate: {
+        $gte: jsforce.SfDate.toDateTimeLiteral(createdDate),
+        $lte: jsforce.SfDate.toDateTimeLiteral(endOfCreatedDate),
+      },
+    })
+  }
+
+  async getApprovedMenteesByDaysAndLocation({
+    daysAgo,
+  }: {
+    daysAgo: '7d' | '14d'
+  }) {
+    const approvedDate = new Date()
+    approvedDate.setDate(approvedDate.getDate() - parseInt(daysAgo)) // Interestingly, parseInt is able to parse 7d and 14d to 7 and 14 respectively
+
+    return this.conProfilesService.findAll({
+      'RecordType.DeveloperName': UserType.MENTEE,
+      Profile_Status__c: ConnectProfileStatus.APPROVED,
+      Profile_First_Approved_At__c: {
+        $eq: jsforce.SfDate.toDateLiteral(approvedDate),
+      },
+    })
+  }
+
+  async getConProfilesWithThreeMonthsOldMentorshipMatches() {
+    const NinetyDaysAgo = new Date().setDate(new Date().getDate() - 90)
+
+    const threeMonthsOldMentorshipMatches =
+      await this.conMentorshipMatchesService.findAll({
+        Status__c: MentorshipMatchStatus.ACCEPTED,
+        Start_Date__c: {
+          $eq: jsforce.SfDate.toDateLiteral(NinetyDaysAgo),
+        },
+      })
+
+    const menteeIds = threeMonthsOldMentorshipMatches.map(
+      (match) => match.props.menteeId
+    )
+    const mentorIds = threeMonthsOldMentorshipMatches.map(
+      (match) => match.props.mentorId
+    )
+
+    if ([...menteeIds, ...mentorIds].length > 0) {
+      return await this.conProfilesService.findAll({
+        'Contact__r.Id': {
+          $in: [...menteeIds, ...mentorIds],
+        },
+      })
+    }
+
+    return []
+  }
+
+  async sendCompleteProfileReminder({
+    userType,
+    email,
+    firstName,
+  }: {
+    userType: UserType
+    email: string
+    firstName: string
+  }) {
+    if (!userType) {
+      throw new Error('User type is required')
+    }
+
+    const sfEmailTemplateName =
+      userType === UserType.MENTOR
+        ? 'Mentor - Profile Completion Reminder'
+        : userType === UserType.MENTEE
+        ? 'Mentee - Profile Completion Reminder'
+        : null
+
+    if (!sfEmailTemplateName) {
+      throw new Error('Invalid user type. Must be either MENTOR or MENTEE')
+    }
+
+    const template = await this.emailTemplatesService.getEmailTemplate(
+      sfEmailTemplateName
+    )
+
+    if (!template) {
+      throw new Error(`Email template not found: ${sfEmailTemplateName}`)
+    }
+
+    const sanitizedHtml = template.HtmlValue.replace(
```

Comment:
Maybe we could even do some kind of test to check for existence of `{{{Recipient.FirstName}}}` here? And throw if not present. That is something that I see relatively easily happening because of user error.

Reply:
I will actually work on these a bit more when I know the script and automation is working correctly. Can we skip this for now?